### PR TITLE
Configure adaptive prop

### PR DESCRIPTION
**Pull request by mod3x**
Hi!. 

Current implementation of adaptive prop (-@media:dark) contains characters that throws an not-valid syntax error on postcss compile step when props are parsed from css files not JSON. 
This PR adds a functionality to configure it.

Thank you.

### CODE REVIEW

File: index.test.js [hunk 1]
```diff
@@ -29,6 +29,21 @@ const MockProps = {
...
+  '--l': '50%',
+  '--size-1': '1rem',
+  '--size-2': '2rem',
+  '--fade-in': 'fade-in .5s ease',
+  '--fade-in-@': '@keyframes fade-in {to { opacity: 1 }}',
+  '--dark': '@custom-media --dark (prefers-color-scheme: dark);',
+  '--text': 'white',
+  '--text-dark': 'black',
+}
+
```
Comment (argyleink) [inline index.test.js:46]: could probably be reduced to just this yeah? less mock data to sift through to find what's being tested?

```suggestion
const MockPropsWithCustomAdaptiveProp = {
  '--text': 'white',
  '--text-dark': 'black',
}
```

File: index.test.js [hunk 2]
```diff
@@ -39,13 +54,13 @@ it('Can jit a single prop', async () => {
   await run(
 `a {
   color: var(--red);
-}`, 
+}`,
```
Comment (argyleink) [inline index.test.js:57]: can you shave these spacing edits from the PR please? will help focus the PR into only what's needing changed.

File: index.test.js [hunk 3]
```diff
@@ -390,7 +405,30 @@ p {
   :root {
     --text: black;
   }
-}`, 
+}`,
   MockProps
   )
 })
+
+it('Can jit a light and dark with custom adaptive prop', async () => {
```
Comment (argyleink) [inline index.test.js:413]:
```suggestion
it('Can jit a light and dark color with a custom adaptive prop parameter', async () => {
```

is this suggestion the same as what you meant?

**Review by argyleink — CHANGES_REQUESTED** [review]
thanks for this!

if i'm understanding correctly, this is introducing a new parameter `adaptive_prop_selector` to help developers avoid a parsing error? are you able to share a stackblitz or codesandbox that reproduce the issue? i havent seen it yet in vite, astro or a vanilla postcss config, so I'm curious and would like to see the full picture 🙂 

all in all though, it looks to be additive and non destructive to the current features. i appreciate you adding a test! 🤘🏻 given the problem scenario makes sense enough, this looks like it can merge soon 👍🏻

**Comment by mod3x** [comment]
@argyleink heres [stackblitz demo](https://stackblitz.com/edit/node-8mv4sx?file=props.css)
package.json contains _postcss_ script to start compilation, thank you

**Comment by mod3x** [comment]
>ty for the repro! looks like it's a bug in the CSS import feature specifically? that props using the naming convention break the CSS parser AND then don't get searched for. yeah?

Youre right.

>i'm almost thinking this means i need to change the syntax for these special props, so that they parse from CSS without any special configuration? it's like i missed this use case entirely in the tests and it's here to show the strategy needs adjusted

It would be cool. I mean its post**css** after all, align it with css spec as much as possible :)

**Review by argyleink — APPROVED** [review]
cool, looks good, ty!

**Pull request merged**